build: re-add Linux ARM64 cross compiler to the builder image tool chains

Between v21.1.0-alpha.3 and v21.1.0-beta.1, the Linux ARM64 compiler disappeared from the `build/builder.sh` builder image. Add the cross compiler back to the image.

When running `build/builder.sh mkrelease linux-gnueabi` on v21.1.0-beta.1 or later, this error is encountered:

```
-- The CXX compiler identification is unknown
-- Check for working C compiler: /usr/local/lib/ccache/aarch64-unknown-linux-gnueabi-cc
-- The CXX compiler identification is GNU 9.3.0
golang.org/x/tools/internal/event
yarn install v1.22.5
-- Check for working C compiler: /usr/local/lib/ccache/aarch64-unknown-linux-gnueabi-cc - broken
CMake Error at /usr/share/cmake-3.17/Modules/CMakeTestCCompiler.cmake:60 (message):
  The C compiler

    "/usr/local/lib/ccache/aarch64-unknown-linux-gnueabi-cc"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: /go/native/aarch64-unknown-linux-gnueabi/geos/CMakeFiles/CMakeTmp

    Run Build Command(s):/usr/bin/make cmTC_a2e14/fast && make[1]: Entering directory '/go/native/aarch64-unknown-linux-gnueabi/geos/CMakeFiles/CMakeTmp'
    /usr/bin/make -f CMakeFiles/cmTC_a2e14.dir/build.make CMakeFiles/cmTC_a2e14.dir/build
    make[2]: Entering directory '/go/native/aarch64-unknown-linux-gnueabi/geos/CMakeFiles/CMakeTmp'
    Building C object CMakeFiles/cmTC_a2e14.dir/testCCompiler.c.o
    /usr/local/lib/ccache/aarch64-unknown-linux-gnueabi-cc   -fPIC    -o CMakeFiles/cmTC_a2e14.dir/testCCompiler.c.o   -c /go/native/aarch64-unknown-linux-gnueabi/geos/CMakeFiles/CMakeTmp/testCCompiler.c
    ccache: error: Could not find compiler "aarch64-unknown-linux-gnueabi-cc" in PATH
    make[2]: *** [CMakeFiles/cmTC_a2e14.dir/build.make:83: CMakeFiles/cmTC_a2e14.dir/testCCompiler.c.o] Error 1
    make[2]: Leaving directory '/go/native/aarch64-unknown-linux-gnueabi/geos/CMakeFiles/CMakeTmp'
    make[1]: *** [Makefile:138: cmTC_a2e14/fast] Error 2
    make[1]: Leaving directory '/go/native/aarch64-unknown-linux-gnueabi/geos/CMakeFiles/CMakeTmp'
```

Epic: CRDB-8229

--- a/build/builder.sh
+++ b/build/builder.sh
@@ -3,7 +3,7 @@
 set -euo pipefail
 
 image=cockroachdb/builder
-version=20210616-013156
+version=20210625-235250
 
 function init() {
   docker build --tag="${image}" "$(dirname "${0}")/builder"

--- a/build/builder/Dockerfile
+++ b/build/builder/Dockerfile
@@ -73,14 +73,14 @@ RUN mkdir -p /usr/local/lib/ccache \
  && ln -s /usr/bin/ccache /usr/local/lib/ccache/x86_64-unknown-linux-gnu-c++ \
  && ln -s /usr/bin/ccache /usr/local/lib/ccache/x86_64-w64-mingw32-cc \
  && ln -s /usr/bin/ccache /usr/local/lib/ccache/x86_64-w64-mingw32-c++ \
- && ln -s /usr/bin/ccache /usr/local/lib/ccache/aarch64-unknown-linux-gnueabi-cc \
- && ln -s /usr/bin/ccache /usr/local/lib/ccache/aarch64-unknown-linux-gnueabi-c++ \
+ && ln -s /usr/bin/ccache /usr/local/lib/ccache/aarch64-unknown-linux-gnu-cc \
+ && ln -s /usr/bin/ccache /usr/local/lib/ccache/aarch64-unknown-linux-gnu-c++ \
  && ln -s /usr/bin/ccache /usr/local/lib/ccache/s390x-ibm-linux-gnu-c++ \
  && ln -s /usr/bin/ccache /usr/local/lib/ccache/s390x-ibm-linux-gnu-cc \
  && ln -s /usr/bin/ccache /usr/local/lib/ccache/x86_64-apple-darwin19-cc \
  && ln -s /usr/bin/ccache /usr/local/lib/ccache/x86_64-apple-darwin19-c++
 
-ENV PATH $PATH:/x-tools/x86_64-unknown-linux-gnu/bin:/x-tools/x86_64-w64-mingw32/bin:/x-tools/aarch64-unknown-linux-gnueabi/bin:/x-tools/s390x-ibm-linux-gnu/bin:/x-tools/x86_64-apple-darwin19/bin
+ENV PATH $PATH:/x-tools/x86_64-unknown-linux-gnu/bin:/x-tools/x86_64-w64-mingw32/bin:/x-tools/aarch64-unknown-linux-gnu/bin:/x-tools/s390x-ibm-linux-gnu/bin:/x-tools/x86_64-apple-darwin19/bin
 
 # Compile GNU sed from source to pick up an unreleased change that buffers
 # output. This speeds up compiles on Docker for Mac by *minutes*.

--- a/build/builder/mkrelease.sh
+++ b/build/builder/mkrelease.sh
@@ -9,7 +9,7 @@
 # Possible configurations:
 #
 #   - amd64-linux-gnu:      amd64, Linux 2.6.32, dynamically link glibc 2.12.2
-#   - arm64-linux-gnueabi:  arm64, Linux 3.7.10, dynamically link glibc 2.12.2
+#   - arm64-linux-gnu:      arm64, Linux 3.7.10, dynamically link glibc 2.12.2
 #   - amd64-darwin:         amd64, macOS 10.9
 #   - amd64-windows:        amd64, Windows 8, statically link all non-Windows libraries
 #   - s390x-linux-gnu:      s390x, Linux 2.6.32, dynamically link glibc 2.12.2
@@ -48,7 +48,7 @@ case "${1-}" in
       SUFFIX=-linux-2.6.32-gnu-amd64
     ) ;;
 
-  ?(arm64-)linux?(-gnueabi))
+  ?(arm64-)linux?(-gnu))
     # Manually set the correct values for configure checks that libkrb5 won't be
     # able to perform because we're cross-compiling.
     export krb5_cv_attr_constructor_destructor=yes
@@ -58,7 +58,7 @@ case "${1-}" in
       XGOOS=linux
       XGOARCH=arm64
       XCMAKE_SYSTEM_NAME=Linux
-      TARGET_TRIPLE=aarch64-unknown-linux-gnueabi
+      TARGET_TRIPLE=aarch64-unknown-linux-gnu
       LDFLAGS="-static-libgcc -static-libstdc++"
       SUFFIX=-linux-3.7.10-gnu-aarch64
     ) ;;

